pkg/controller/monitor/monitor_controller.go

```diff
@@ -265,6 +294,64 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, err
 	}
 
+	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.client)
+	if err != nil {
+		log.Error(err, "Error reading ManagementClusterConnection")
+		r.status.SetDegraded("Error reading ManagementClusterConnection", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// In managed clusters with certificate management disabled, successful reconciliation of non-NetworkPolicy resources
+	// in the monitor controller ensures that NetworkPolicy is reconcilable (by enabling* the creation of containing Tier
+	// and License). Therefore, to prevent a chicken-and-egg scenario, we only reconcile non-NetworkPolicy resources once
+	// we can confirm that all requirements to reconcile NetworkPolicy have been met.
+	//
+	// * In managed clusters with certificate management disabled, the License can only be pushed once Guardian is deployed,
+	//   which depends on the Prometheus TLS secret created by monitor. As always, the Tier can only be created once the
+	//   License is available.
+	includeV3NetworkPolicy := true
+	monitorReconcilesV3NetworkPolicyRequirements := managementClusterConnection != nil && !certificateManager.KeyPair().UseCertificateManagement()
+	if monitorReconcilesV3NetworkPolicyRequirements {
+		// Only wait for and render NetworkPolicy once all NetworkPolicy requirements are met.
+		// Do not degrade if tier has not been created.
+		if utils.IsV3NetworkPolicyReconcilable(ctx, r.client, networkpolicy.TigeraComponentTierName) {
+			if !r.tierWatchReady.IsReady() {
+				r.status.SetDegraded("Waiting for Tier watch to be established", "")
+				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+			}
+
+			if !r.policyWatchesReady.IsReady() {
+				r.status.SetDegraded("Waiting for NetworkPolicy watches to be established", "")
+				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+			}
+		} else {
+			includeV3NetworkPolicy = false
+		}
+	} else {
+		// Wait for and render NetworkPolicy as usual.
+		if !r.tierWatchReady.IsReady() {
+			r.status.SetDegraded("Waiting for Tier watch to be established", "")
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		}
+
+		if !r.policyWatchesReady.IsReady() {
+			r.status.SetDegraded("Waiting for NetworkPolicy watches to be established", "")
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		}
+
+		// Ensure the allow-tigera tier exists, before rendering any network policies within it.
+		if err := r.client.Get(ctx, client.ObjectKey{Name: networkpolicy.TigeraComponentTierName}, &v3.Tier{}); err != nil {
```

Comment:
I find it a little confusing that in some places, we query the Tier directly to determine if it exists, and in some cases we essentially do the same by calling `utils.IsV3NetworkPolicyReconcilable(ctx, r.client, networkpolicy.TigeraComponentTierName)`

The obvious difference seems to be that the former has more detailed error handling, whereas the latter has no error handling (errors querying the tier are effectively ignored, and assumed to be 404 errors). I suspect we do actually need to handle errors when calling `IsV3NetworkPolicyReconcilable` though.